### ISSUE TITLE
creates too many threads

### ISSUE THREAD

**Issue by sunmisc** [ajgeiss0702/ajLeaderboards#35]
I can't understand why the plugin creates so MUCH expensive os threads for blocking I/O.

OS threads are not the best option for blocking I/O, but if anything, you should use them to a limited extent. It is quite enough to have a number of threads comparable to the hardwares threads per subsystem, or even better, per JVM.

In the config I specified 2 threads (by default 70 is a lot! ~70mb), but why the pool has swollen to 31 threads and continues to grow, which naturally clogs the main scheduler, especially bad will be weak hardwares

You can use Virtual Threads for Jdk 21+, or you can use ForkJoinPool with async mode
![image](https://github.com/user-attachments/assets/f7241fe0-e078-49ab-b58f-0d54db219c6b)

**Comment by sunmisc** [comment]
![image](https://github.com/user-attachments/assets/fffb8d2f-3168-40c3-b933-2cf9851f00f7)
If spark isn't wrong (and judging by the fact that memory ran out and tps dropped to 5, it is) it's Very Bad, the server literally went down

**Comment by ajgeiss0702** [comment]
This is not the place to get support

**Comment by ajgeiss0702** [comment]
> [@ajgeiss0702](https://github.com/ajgeiss0702) returning to this issue as it seems you have no intention of ever optimizing this. The user above is not asking for support, he is clearly stating in plain English a serious problem with your plugin that causes an insane amount of threads to be created and causes server crashes. This issue alone is causing people to migrate to plugins like Topper because you won't listen to reason. Please, stop ignoring the community on this. I shouldn't have to increase limits on my linux machine to facilitate your plugin. Server owners on managed hosts don't even have that option.

I’ve helped dozens of people fix this issue on their server. ajlb was the cause 2-3 times.

If you really don’t like the way ajlb handles threads, it is easily changeable in the config.

You would have known this if you went to support. Which is why I say to go to support
